File: server/konpeko.py
from concurrent import futures
import logging

import grpc

import konpeko_pb2
import konpeko_pb2_grpc


class Pekora(konpeko_pb2_grpc.PekoraServicer):
    def IdolGreeting(self, request, context):
        return konpeko_pb2.GreetReply(message="%s さん、こんぺこ～！" % request.name)


def serve():
    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
    konpeko_pb2_grpc.add_PekoraServicer_to_server(Pekora(), server)
    server.add_insecure_port("[::]:50051")
    server.start()
    server.wait_for_termination()


if __name__ == "__main__":
    logging.basicConfig()
    serve()

File: server/konpeko.go
// Package main implements a server for Greeter service.
package main

import (
	"context"
	"log"
	"net"

	"google.golang.org/grpc"
	pb "github.com/yumechi/grpc_suburi/server/protos"
)

const (
	port = ":50051"
)

// server is used to implement helloworld.GreeterServer.
type server struct {
	pb.UnimplementedPekoraServer
}

// SayHello implements helloworld.GreeterServer
func (s *server) IdolGreeting(ctx context.Context, in *pb.GreetRequest) (*pb.GreetReply, error) {
	log.Printf("Received: %v", in.GetName())
	return &pb.GreetReply{Message: in.GetName() + "さん、こんぺこ～！"}, nil
}

func main() {
	lis, err := net.Listen("tcp", port)
	if err != nil {
		log.Fatalf("failed to listen: %v", err)
	}
	s := grpc.NewServer()
	pb.RegisterPekoraServer(s, &server{})
	if err := s.Serve(lis); err != nil {
		log.Fatalf("failed to serve: %v", err)
	}
}